### PR TITLE
Nodejs/week2

### PR DESCRIPTION
**Pull request by m-elshawa**

### CODE REVIEW

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+const router = express.Router();
+
+const meals = require("../data/meals.json");
+
+// 12 tests passed, 6 failed. Can't get my code to work 
+
+router.get("/", async (request, response) => {
+  try {
+    console.log(request.query.maxPrice);
+    response.send(meals);
```
Comment (Laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:12]: Hm, is it just a test route?

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+router.get("/large-meals", async (request, response) => {
+  largeMeals = meals.filter((largeMeal) => {
+    return largeMeal.maxNumberOfGuests > 3;
+  });
+  response.json(largeMeals);
+});
+
+router.get("/meal", async (request, response) => {
+  response.send("Meal Sharing Web App");
+});
```
Comment (Laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:41]: It seems like you are missing big part of this week's homework here. Look at the description of the homework, and you'll see what routes and query parameters you should implement.

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      console.log(reservations);
+      for (let x = 0; x < reservations.length; x++) {
+        
+          //  console.log(reservations[x].name);
+           if (request.params.id == reservations[x].id) {
+            r = reservations[x];
+           }
+           
+
+      }
```
Comment (Laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:19]: Good

File: nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  router.get("/:id", async (request, response) => {
+    let eachReview = undefined;
+    // console.log(reviews);
+    for (let x = 0; x < reviews.length; x++) {
+      if (request.params.id == reviews[x].id) {
+        eachReview = reviews[x];
+      }
+    }
+    response.send(eachReview);
```
Comment (Laila1222) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reviews-router.js:23]: Good

File: nodejs/week2/homework/warmup/index.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("nodejs week2 homework"));
+
```
Comment (Laila1222) [inline nodejs/week2/homework/warmup/index.js:5]: Missing exercise

**Review by Laila1222 — CHANGES_REQUESTED** [review]
Hey :) 
It seems like you are missing a big part of the homework here. 
If you need any help for moving forward, you can ask for help in the different slack channels.

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js (hunk 2 above)
Comment (m-elshawa) [reply nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:41]: well, I'm still trying to get all tests succeed, I get 12 passed and 6 failed. and yet I can't figure it out.

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js (hunk 1 above)
Comment (m-elshawa) [reply nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:12]: unfortunately I don remember, I've been stuck here for months. and in my previous class there was a lot of confusion. so maybe I did it to make sure the route is working.

**Pull request merged**